spring-cloud-dataflow-rest-client/src/main/java/org/springframework/cloud/dataflow/rest/client/DataFlowTemplate.java

```diff
@@ -163,8 +163,13 @@ public DataFlowTemplate(URI baseURI, RestTemplate restTemplate) {
 		final RootResource resourceSupport = restTemplate.getForObject(baseURI, RootResource.class);
 
 		if (resourceSupport != null) {
-			String serverRevision = resourceSupport.getApiRevision() != null
-					? resourceSupport.getApiRevision().toString() : "[unknown]";
+			if (resourceSupport.getApiRevision() == null) {
+				throw new IllegalStateException("Incompatible version of Data Flow server detected.\n"
```

Comment:
That's what I'm trying to do, we don't have a specific download page to direct to so it has to be very generic. If we know the version it is different and we can point to the about page.